staging/src/k8s.io/kube-aggregator/pkg/controllers/openapi/controller.go

```diff
@@ -70,32 +71,42 @@ func NewAggregationController(downloader *aggregator.Downloader, openAPIAggregat
 	return c
 }
 
-// Run starts OpenAPI AggregationController
+// Run is a legacy wrapper that starts the controller.
+//
+//logcheck:context // RunWithContext should be used instead of Run in code which supports contextual logging.
 func (c *AggregationController) Run(stopCh <-chan struct{}) {
-	defer utilruntime.HandleCrash()
+	c.RunWithContext(wait.ContextForChannel(stopCh))
+}
+
+// RunWithContext starts OpenAPI AggregationController and blocks until the context is cancelled.
+func (c *AggregationController) RunWithContext(ctx context.Context) {
+	defer utilruntime.HandleCrashWithContext(ctx)
 	defer c.queue.ShutDown()
 
-	klog.Info("Starting OpenAPI AggregationController")
-	defer klog.Info("Shutting down OpenAPI AggregationController")
+	logger := klog.FromContext(ctx)
+	logger.Info("Starting OpenAPI AggregationController")
+	defer logger.Info("Shutting down OpenAPI AggregationController")
 
-	go wait.Until(c.runWorker, time.Second, stopCh)
+	go wait.UntilWithContext(ctx, c.runWorker, time.Second)
```

Comment:
Seems like we can take this opportunity to remove the extra goroutine call and remove the `<-ctx.Done()` wait?

```suggestion
	wait.UntilWithContext(ctx, c.runWorker, time.Second)
```

Similarly for openapiv3